src/App.php

```diff
@@ -1179,203 +1021,6 @@ public function proc_run($command, $args)
 		proc_close($resource);
 	}
 
-	/**
-	 * @brief Returns the system user that is executing the script
-	 *
-	 * This mostly returns something like "www-data".
-	 *
-	 * @return string system username
-	 */
-	private static function getSystemUser()
-	{
-		if (!function_exists('posix_getpwuid') || !function_exists('posix_geteuid')) {
-			return '';
-		}
-
-		$processUser = posix_getpwuid(posix_geteuid());
-		return $processUser['name'];
-	}
-
-	/**
-	 * @brief Checks if a given directory is usable for the system
-	 *
-	 * @param      $directory
-	 * @param bool $check_writable
-	 * @return boolean the directory is usable
-	 * @throws Exception
-	 */
-	public static function isDirectoryUsable($directory, $check_writable = true)
-	{
-		if ($directory == '') {
-			Core\Logger::log('Directory is empty. This shouldn\'t happen.', Core\Logger::DEBUG);
-			return false;
-		}
-
-		if (!file_exists($directory)) {
-			Core\Logger::log('Path "' . $directory . '" does not exist for user ' . self::getSystemUser(), Core\Logger::DEBUG);
-			return false;
-		}
-
-		if (is_file($directory)) {
-			Core\Logger::log('Path "' . $directory . '" is a file for user ' . self::getSystemUser(), Core\Logger::DEBUG);
-			return false;
-		}
-
-		if (!is_dir($directory)) {
-			Core\Logger::log('Path "' . $directory . '" is not a directory for user ' . self::getSystemUser(), Core\Logger::DEBUG);
-			return false;
-		}
-
-		if ($check_writable && !is_writable($directory)) {
-			Core\Logger::log('Path "' . $directory . '" is not writable for user ' . self::getSystemUser(), Core\Logger::DEBUG);
-			return false;
-		}
-
-		return true;
-	}
-
-	/**
-	 * @param string $cat     Config category
-	 * @param string $k       Config key
-	 * @param mixed  $default Default value if it isn't set
-	 *
-	 * @return string Returns the value of the Config entry
-	 */
-	public function getConfigValue($cat, $k, $default = null)
```

Comment:
Can't we keep this method as a wrapper for `$this->getConfig()->get($cat, $k, $default`?

Reply:
Yeah, would be better than the `getConfig()->get()` construct

Reply:
There are just three classes, which would using it:
1. https://github.com/friendica/friendica/blob/0939529450056747635aef3c0580cee15d544f6e/src/Core/Console/Typo.php#L46

2. https://github.com/friendica/friendica/blob/0939529450056747635aef3c0580cee15d544f6e/src/Core/Console/AutomaticInstallation.php#L103-L106

3. https://github.com/friendica/friendica/blob/0939529450056747635aef3c0580cee15d544f6e/src/Core/Console/Config.php#L127-L129

Do we really need it?

Reply:
I guess not, now that you have the ConfigCache mechanism.